src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1348,6 +1351,7 @@ public function getMethod()
                     $this->method = strtoupper($this->request->get('_method', $this->query->get('_method', 'POST')));
                 }
             }
+            self::$httpMethodCacheIsValid = true;
```

Comment:
this is broken anyway. If you have multiple Request objects, the cache will be invalidated only for the first one on which ``getMethod`` is called, but not for other ones